### PR TITLE
Enable Custom UI links

### LINKED ISSUE #1792: Support markdown links in custom UI

### PR DESCRIPTION
**Pull request by GuillaumeRx**
## **Description**

This adds the ability to use MarkDown links in custom UI.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/551
Fixes https://github.com/MetaMask/snaps/issues/1792

## **Manual testing steps**

1. Add a MarkDown link in a custom UI text.
2. Look at the link in the dialog window.

### Custom UI
image

image

### Notifications
image

### TX Insight

image

### Phishing link detected
 
https://github.com/MetaMask/metamask-extension/assets/13910212/778f75a2-8a32-47fd-8d91-222f12f538fd

### Invalid URL

https://github.com/MetaMask/metamask-extension/assets/13910212/55716187-6ca8-4ba6-8df2-14aff71dd1d2

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -2124,9 +2124,18 @@ export default class MetamaskController extends EventEmitter {
           this.controllerMessenger,
           'PhishingController:maybeUpdateState',
         ),
-        isOnPhishingList: (origin) =>
-          this.controllerMessenger.call('PhishingController:testOrigin', origin)
-            .result,
+        isOnPhishingList: (origin) => {
+          const { usePhishDetect } =
+            this.preferencesController.store.getState();
+          if (!usePhishDetect) {
```
Comment (FrederikBolding) [inline app/scripts/metamask-controller.js:2130]: Let's do the same for `maybeUpdatePhishingList`

File: ui/components/app/metamask-template-renderer/safe-component-list.js [hunk 2]
```diff
@@ -12,7 +12,12 @@ import TextField from '../../ui/text-field';
 import ConfirmationNetworkSwitch from '../../../pages/confirmation/components/confirmation-network-switch';
 import UrlIcon from '../../ui/url-icon';
 import Tooltip from '../../ui/tooltip/tooltip';
-import { AvatarIcon } from '../../component-library';
+import {
+  AvatarIcon,
+  ///: BEGIN:ONLY_INCLUDE_IN(snaps)
+  ButtonLink,
```
Comment (FrederikBolding) [inline ui/components/app/metamask-template-renderer/safe-component-list.js:18]: Why is this needed?

File: ui/pages/notifications/index.scss [hunk 3]
```diff
@@ -105,10 +105,6 @@
       flex-direction: column;
       margin-left: 12px;
 
-      &__message {
```
Comment (FrederikBolding) [inline ui/pages/notifications/index.scss:108]: Do we not need this anymore?

File: ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js [hunk 4]
```diff
@@ -16,22 +22,65 @@ const Paragraph = (props) => (
...
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [redirectUrl, setRedirectUrl] = useState(undefined);
+
   if (markdown === false) {
     return <Paragraph>{children}</Paragraph>;
   }
 
+  const handleLinkClick = (url) => {
+    setRedirectUrl(url);
+    setIsModalOpen(true);
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js:48]: Wondering if it makes sense to just use `reidrectUrl` to determine if the modal should be opened?

**Review by FrederikBolding — COMMENTED** [review]
When using this with a dialog the dialog is hidden behind the newly opened page. Is this intentional? We should be sure we are fine with that.

Also @eriknson should review this PR

**Comment by GuillaumeRx** [comment]
> When using this with a dialog the dialog is hidden behind the newly opened page. Is this intentional? We should be sure we are fine with that.
> 
> Also @eriknson should review this PR

It's just the default behavior of the browser, we can't really control this.

@eriknson approved the designs :)

File: ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js (hunk 4 above)
Comment (GuillaumeRx) [reply ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js:48]: addressing this 👍

File: ui/pages/notifications/index.scss (hunk 3 above)
Comment (GuillaumeRx) [reply ui/pages/notifications/index.scss:108]: Nope, our MarkDown renderer handles it :)

**Review by FrederikBolding — APPROVED** [review]

**Review by david0xd — APPROVED** [review]
